server/src/core/payment/tests/transaction-service.test.ts

```diff
@@ -0,0 +1,110 @@
+import { createDbUtils, expectAsyncAppError } from '../../test-util';
+import { transactions } from './fixtures';
+import { Transactions } from '../transaction-service';
+import { EventBus } from '../../event';
+import { PaymentProviders } from '../provider-registry';
+import { PaymentProviderRegistry, PaymentProvider, SendFundsResult } from '../types';
+
+const DB = '_social_relief_transaction_service_tests_';
+const COLLECTION = 'transactions';
+
+describe('TransactionService tests', () => {
+  const dbUtils = createDbUtils(DB, COLLECTION);
+  let paymentProviders: PaymentProviderRegistry;
+  let testPaymentProvider: PaymentProvider;
+
+  beforeAll(async () => {
+    await dbUtils.setupDb();
+  });
+
+  beforeEach(async () => {
+    await dbUtils.resetCollectionWith(transactions);
+  });
+
+  afterAll(async () => {
+    await dbUtils.tearDown();
+  });
+
+  beforeEach(() => {
+    paymentProviders = new PaymentProviders();
+    testPaymentProvider = {
+      name: () => 'testPaymentProvider',
+      requestPaymentFromUser: jest.fn(),
+      handlePaymentNotification: jest.fn(),
+      getTransaction: jest.fn(),
+      sendFundsToUser: jest.fn()
+    };
+    
+    paymentProviders.register(testPaymentProvider);
+    paymentProviders.setPreferredForSending(testPaymentProvider.name());
+    paymentProviders.setPreferredForReceiving(testPaymentProvider.name());
+  });
+
+  function createService() {
+    const service = new Transactions(dbUtils.getDb(), {
+      eventBus: new EventBus(),
+      paymentProviders
+    });
+
+    return service;
+  }
+
+  describe('getUserBalance', () => {
+    test('returns 0 if user has no transactions', async () => {
+      const user = 'userWithNoTransactions';
+      var service = createService();
```

Comment:
Any particular reason why you're using `var` instead of `const`

Reply:
I didn't notice I user `var`. Thanks.